dogfood.sh

```diff
@@ -0,0 +1,28 @@
+if [ -z "$ZSH_VERSION" ]; then
```

Comment:
NIT: should probably have `#!/usr/bin/env bash` at the top line of this script.

Reply:
Also, I don't see in other scripts in the repo us checking if you are using Bash or Zsh, is there a reason why we need that here?

Reply:
This is meant to be used like `source dogfood.sh`, so that wouldn't be appropriate AFAIU.

Reply:
`${BASH_SOURCE[0]}` is not supported in `zsh`.

Reply:
I see, I think that having it be a `.sh` will be misleading then if the intent is to do source dogfood.sh. Just curious, why not make it an actual bash script that can be executed? Here is precedent of a very similar use cas in the sdk repo: https://github.com/dotnet/sdk/blob/main/eng/dogfood.sh

Reply:
If the script is run as `/foo/dogfood.sh` then the changes to the environment will be limited to that script. The documentation in sdk also suggests using `source dogfood.sh`:

https://github.com/dotnet/sdk/blob/9e277c9ce470d35ee4ebe751a5629be8688f5037/documentation/project-docs/developer-guide.md?plain=1#L108

When using `source ..` we are using the current shell to run the commands so the shebang should be not be needed, AFAIU.